### ISSUE TITLE
Failed to get requirements from a jar file

### ISSUE THREAD

**Issue by ereticul** [serenity-bdd/serenity-core#39]
Hey there, 

I've tried the following:
FileSystemRequirementsTagProvider fileSystemRequirementsTagProvider =
                new FileSystemRequirementsTagProvider("com/forrest/gump/integrationtest/cucumber/");
 fileSystemRequirementsTagProvider.getRequirements();

and getRootDirectoryPaths() return the following path:
file:/usr/bin/apache-tomcat-7.0.53/webapps/ROOT/WEB-INF/lib/cucumber-demo-project-0.6.3.12-SNAPSHOT.jar!/com/forrest/gump/integrationtest/cucumber/

and yes, if new File(rootDirectoryPath).exists => then false, and failed to get the requirements.

Should it be treated as an enhancement of getting requirements from the JAR file? or is this an issue actually? Or there is another way of reading the requirements? Please adivise.

Andy.

**Comment by ereticul** [comment]
I've just evaluated that: allRequirements.addAll(loadCapabilitiesFrom(rootDirectory.listFiles(thatAreDirectories()))); 
and I got successfully the requiements based on above root directory path.

@John, is it necessary this check of existence of file? Please have a look as soon as you can. Because it would fail to get the requirements, and no 'capability' tag would be generated as I understand.

Thank you.

**Comment by wakaleo** [comment]
Hi Andy,

Reading requirements from JAR files is not currently supported, so it would
be an enhancement. Out of curiosity, why do you need to read a requirements
structure from a dependency?

- John.

On 26 March 2015 at 08:00, ereticul <notifications@github.com> wrote:

> Hey there,
>
> I've tried the following:
> FileSystemRequirementsTagProvider fileSystemRequirementsTagProvider =
> new
> FileSystemRequirementsTagProvider("com/forrest/gump/integrationtest/cucumber/");
> fileSystemRequirementsTagProvider.getRequirements();
>
> and getRootDirectoryPaths() return the following path:
>
> file:/usr/bin/apache-tomcat-7.0.53/webapps/ROOT/WEB-INF/lib/cucumber-demo-project-0.6.3.12-SNAPSHOT.jar!/com/forrest/gump/integrationtest/cucumber/
>
> and yes, if new File(rootDirectoryPath).exists => then false, and failed
> to get the requirements.
>
> Should it be treated as an enhancement of getting requirements from the
> JAR file? or is this an issue actually? Or there is another way of reading
> the requirements? Please adivise.
>
> Andy.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/serenity-bdd/serenity-core/issues/39>.
>



-- 
___________________________________________________
John Smart | Wakaleo Consulting  |  +61 407 247 642
Optimizing your software development process
http://www.wakaleo.com  |  john.smart@wakaleo.com
___________________________________________________

The dates for the 2014 BDD workshops have been scheduled! Check out our
upcoming *BDD/TDD Master classes*
<http://wakaleo.com/training/testing-and-tdd-for-java-developers> and
our *Advanced
BDD Requirements Workshops
<http://wakaleo.com/training/advanced-agile-requirements-workshop>*, coming
soon to *Sydney
<http://wakaleo.com/training/testing-and-tdd-for-java-developers/public-courses>
*
and *Melbourne
<http://wakaleo.com/training/testing-and-tdd-for-java-developers/public-courses>!*
___________________________________________________

**Comment by ereticul** [comment]
Hi John, 

I've implemented a lightweight runner of executing the cucumber scenarios, it's alike the story runner of JBehave, but for Cucumber-JVM. I noticed if I can read the requirements then I would be able to build up a UI tree component based on requirement model, isn't it? and not only at the moment there is a shortcoming, while executing the cucumber-jvm + serenity core, the test outcome is generated with no tag like 'capability', so below won't work and the report won't contain the capability, isn't it?

serenity.requirements.dir=com/forrest/gump/integrationtest/cucumber/serenity/junit
serenity.requirement.types=capability,feature,story

Andy.
